### PR TITLE
Template field support for configmaps in the KubernetesPodOperator

### LINKED ISSUE #36888: Support for configmaps (as env var & volume) in the template fields in the KubernetesPodOperator
### Description

Currently configmaps are not supported in the template fields in the KubernetesPodOperator. It will be helpful to add this support to dynamically attach configmaps to a pod 

### Use case/motivation

Currently configmaps are not supported in the template fields in the KubernetesPodOperator. It will be helpful to add this support to dynamically attach configmaps to a pod 

**I have made the code changes locally and also added unittests. I have also run all the Kubernetes provider unit tests after the changes**

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by vizeit**
---
Currently configmaps are not supported in the template fields in the KubernetesPodOperator. It will be helpful to add this support to dynamically attach configmaps to a pod. This change adds support for template field for configmaps

closes: [#36888](https://github.com/apache/airflow/issues/36888)

### CODE REVIEW

**Review by dirrao — APPROVED** [review]
Nice work. LGTM.

File: newsfragments/36922.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Support for configmaps in the template fields in the KubernetesPodOperator
```
Comment (eladkal) [inline newsfragments/36922.feature.rst]: Please remove this file.
This change is in providers we dont need newsfragment for providers :)

**Comment by eladkal** [comment]
We need to make sure this PR doesn't add extra work for https://github.com/apache/airflow/issues/36484 
cc @romsharon98

**Comment by romsharon98** [comment]
> We need to make sure this PR doesn't add extra work for #36484 cc @romsharon98

Checked it, looks good.

**Review by romsharon98 — APPROVED** [review]

**Review by hussein-awala — APPROVED** [review]
Looks good, LGTM

**Review by eladkal — APPROVED** [review]

**Pull request merged**